lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -352,6 +367,26 @@ private void storeHmrcInheritedIdentity(
         }
     }
 
+    private boolean isHmrcInheritedIdentityWithStrongerVotPresent(
+            SignedJWT incomingInheritedIdentity, String userId) throws CredentialParseException {
+        try {
+            var vcStoreItem =
+                    verifiableCredentialService.getVcStoreItem(HMRC_MIGRATION_CRI, userId);
+            SignedJWT existingInheritedIdentity = SignedJWT.parse(vcStoreItem.getCredential());
+
+            var existingInheritedIdentityVot =
+                    userIdentityService.getVot(existingInheritedIdentity);
+            var incomingInheritedIdentityVot =
+                    userIdentityService.getVot(incomingInheritedIdentity);
+
+            return existingInheritedIdentityVot.compareTo(incomingInheritedIdentityVot) > 0;
+        } catch (JsonProcessingException | ParseException | IllegalArgumentException e) {
+            throw new CredentialParseException(
+                    "Encountered a parsing error while attempting to parse credentials: "
+                            + e.getMessage());
+        }
+    }
```

Comment:
Here's the important stuff ^. We check if there's a stored inherited identity with a stronger vot already